custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -0,0 +1,194 @@
+import datetime
+
+from simplejson import JSONDecodeError
+
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.repeaters.repeater_generators import RegisterGenerator, CaseRepeaterJsonPayloadGenerator
+from custom.enikshay.case_utils import get_person_case_from_episode
+from custom.enikshay.integrations.nikshay.repeaters import NikshayRegisterPatientRepeater
+from custom.enikshay.integrations.nikshay.field_mappings import (
+    gender_mapping, occupation, episode_site,
+    treatment_support_designation, patient_type_choice,
+    disease_classification
+)
+from custom.enikshay.integrations.ninetyninedots.repeater_generators import _update_person_case
+
+ENIKSHAY_ID = 8
+
+
+@RegisterGenerator(NikshayRegisterPatientRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayRegisterPatientPayloadGenerator(CaseRepeaterJsonPayloadGenerator):
+    def get_payload(self, repeat_record, episode_case):
+        """
+        mandatory_fields for Nikshay Registration API as of now
+        scode
+        dcode
+        pname
+        pgender
+        paddress
+        pmob
+        ptbyr
+        pregdate
+        cname
+        caddress
+        cmob
+        dcpulmunory # Pending
+        dotname
+        dotdesignation
+        dotmob
+        dotpType
+        dotcenter # Pending
+        regBy # Used date_of_diagnosis instead of date_of_registration
+        IP_From # Pending
+        Local_ID
+        tcode # Pending
+        page
+        poccupation
+        dotphi
+        Ptype
+        pcategory # Pending
+        Source
+        """
+        # properties skipped with example values
+        # "plandline": "N/A",
+        # "clandline": "N/A",
+        # "dotlandline": "N/A",
+        # "cvisitedby": "N/A",
+        # "cvisitedDate": "N/A",
+        # "dotcenter": "N/A",
+        # "dotmoname": "N/A",
+        # "mosign": "N/A",
+        # "mosigndonedate": "N/A",
+        # "atbtreatment": "N/A",
+        # "atbduration": ,
+        # "atbsource": "G",
+        # "atbregimen": "Manish-Regim",
+        # "atbyr": "2015",
+        # "dcpulmunory": "P",
+        # "IP_Address": "mk-ip-address",
+        # "IP_From": "127.0.0.1",
+        person_case = get_person_case_from_episode(repeat_record.domain, episode_case.get_id)
+
+        episode_case_properties = episode_case.case_properties()
+        person_case_properties = person_case.case_properties()
+
+        properties_dict = {
+            "regBy": self.repeater.username,
+            "Local_ID": person_case.get_id,
+            "Source": ENIKSHAY_ID
+        }
+        properties_dict.update(_get_person_case_properties(person_case_properties))
+        properties_dict.update(_get_episode_case_properties(episode_case_properties))
+        return json.dumps(properties_dict)
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # A success would be getting a nikshay_id for the patient
+        # without it this would actually be a failure
```

Comment:
hah. so we can't tell whether it is a successful response just through the HTTP status code?